message_sender/serializers.py

```diff
@@ -22,6 +22,36 @@ class Meta:
             'helper_metadata', 'created_at', 'updated_at')
 
 
+class JunebugInboundSerializer(serializers.HyperlinkedModelSerializer):
+    """
+    Maps fields from Junebug onto fields expected by the Inbound model.
+    """
+    reply_to = serializers.CharField(source='in_reply_to')
+    to = serializers.CharField(source='to_addr')
+    channel_id = serializers.CharField(source='transport_name')
+    channel_data = serializers.JSONField(source='helper_metadata')
```

Comment:
For the `continue_session` and `session_event` fields, on a vumi message they're in the root of the message, but for a Junebug message, they're in `channel_data`: https://github.com/praekelt/junebug/blob/develop/junebug/utils.py#L51-L54 . I think the `session_event` field is the important one that you're interested in for the concurrency limiter?

That poses a problem since I don't think the `session_event` will go through the vumi message serializer? (I'm not sure why it wasn't added, since it's a required field for a vumi message https://github.com/praekelt/vumi/blob/develop/vumi/message.py#L283-L295) I'm wondering if we should make the vumi serializer add those two fields to the `helper_metadata` dict, to make it more like this serializer; or if we should add those fields to the model and the vumi serializer, and extract them from `channel_data` in this serializer?